backend/dataall/core/environment/services/environment_service.py

```diff
@@ -81,7 +81,7 @@ def create_environment(session, uri, data=None):
             resource_prefix=env.resourcePrefix,
         ).build_compliant_name()
 
-        if not data.get('EnvironmentDefaultIAMRoleArn'):
+        if not data.get('EnvironmentDefaultIAMRoleName'):
```

Comment:
@anushka-singh I think there is some commits-conflict issues in this PR. From what I am seeing this PR overrides the [PR](https://github.com/awslabs/aws-dataall/pull/749/files#diff-87a10df353d26bf2d94cf1d6dbcfe8a58a129c7f0505f62539c7131036a2d984) that handles roles by arn instead of by name

Reply:
I just checked the logs of the PR branch and the last time it got commits from `main` or `v2m1m0` was on Sept 16th. Changes done by @noah-paige in the env-role-arn PR were merged on Sept 26th. That's why they are missing. We might face other conflict issues/overrides besides this one, so the best will be to do a merge of `origin/v2m1m0` into your working branch `anushka/v2m1m0`. You will probably face some conflict issues, if you want @noah-paige or I can do the merge with you. And of course if you have a better way to do it, we can do it in a different way. Git-ing is full possibilities